packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/add/dedicatedCloud-datacenter-add.controller.js

```diff
@@ -38,33 +42,86 @@ export default class {
           )}: ${err.message || ''}`,
           'danger',
         );
+      });
+  }
+
+  calculateEstimate() {
```

Comment:
The function is hard to read. Maybe, you should add some comments to ease comprehension.
Furthermore, according to me, price calculation is only business logic so it could be moved to a dedicated service